outputs.tf

```
@@ -1,19 +1,19 @@
 output "dynamodb_table_arn" {
   description = "ARN of the DynamoDB table"
-  value       = element(concat(aws_dynamodb_table.this.*.arn, [""]), 0)
+  value       = element(concat(local.table_in_use.*.arn, [""]), 0)
```

these could become:
```suggestion
  value       = try(aws_dynamodb_table.this[0].arn, aws_dynamodb_table.autoscaled[0].arn, "")
```

since its only possible to have non-autoscaled, autoscaled, or not-created

---

Cool, I like it, looks cleaner